### DISCUSSION TITLE
Input is disabled for primary key

### DISCUSSION THREAD

**Discussion by fishnux** [directus/directus#12273] [Archive]
I'm not sure if this is intentional or not, but when creating a new row I can't input the number on the primary key field (which is usually the ID), as the input is disabled by the HTML tag "disabled"

In my use-case, I actually want to be able to input arbitrary IDs

### To Reproduce

Create a table with an integer primary key, with auto-increment*, and then use Directus on top of such database

* auto-increment is not needed for my use-case, although the existing database has it

<details>
<summary>More Info</summary>

### What version of Directus are you using?

9.7.0

### What version of Node.js are you using?

16.14.2

### What database are you using?

PostgreSQL

### What browser are you using?

Brave (Chromium-based)

### What operating system are you using?

MacOS

### How are you deploying Directus?

NodeJS on remote host (Plesk)

</details>

**Comment by azrikahar** [comment]
> I'm not sure if this is intentional or not, but when creating a new row I can't input the number on the primary key field (which is usually the ID), as the input is disabled by the HTML tag "disabled"

This is indeed intentional, and you can refer to the answer on a similar past discussion here which elaborates more on it: https://github.com/directus/directus/discussions/10354#discussioncomment-1810786

> In my use-case, I actually want to be able to input arbitrary IDs

By taking the linked answer above into account here (manual string ID being the only editable one), would you mind expanding on the needs of your use case?

**Reply by fishnux** [reply]
Hi,

Thanks for the link

In my use-case, I want to create items that may not following a sequential ID - they are unique integers although not sequential.

Actually, I just took out AUTO_INCREMENT on the database, and the field is no longer disabled :)

**Reply by azrikahar** [reply]
> they are unique integers although not sequential.

That makes sense. Removing `AUTO_INCREMENT` sounds like an actual solution from database perspective even if Directus is not in the picture here. Glad it worked out 👍
